### PR TITLE
ddl: Add dbInfo and tblInfo in some operations of DDL

### LINKED ISSUE #1842: should we add tableInfo into ddl job?
I think we can add the associated tableInfo into dll job (addColumn,dropColumn,addIndex, dropIndex) before we place it into history ddl job queue.

The cause is  TiDB-Binlog need to compute the new schema base on ddl job.
I don't think it can guaranteed completely correct, so load from TiDB maybe is the best way.

@shenli @coocood @iamxy

### PR DESCRIPTION
**Pull request by zimulala**
Fix #1842

### CODE REVIEW

File: ddl/reorg.go [hunk 1]
```diff
@@ -242,14 +242,15 @@ func (d *ddl) delKeysWithPrefix(prefix kv.Key, jobType JobType, job *model.Job,
...
 
-// addFinishInfo adds schema version and table information that are used for binlog.
-// tblInfo is added in the following operations: add column, drop column, add index, drop index.
-func addFinishInfo(job *model.Job, ver int64, tblInfo *model.TableInfo) {
-	if tblInfo == nil {
-		job.Args = []interface{}{ver}
+// addFinishInfo adds schema version and information (schema or table information) that are used for binlog.
+// dbInfo is added in the following operations: create database, drop database.
+// And the other operations add tblInfo.
+func addFinishInfo(job *model.Job, ver int64, dbInfo *model.DBInfo, tblInfo *model.TableInfo) {
```
Comment (coocood) [inline ddl/reorg.go]: Rename to `addHistoryInfo` is better.
The job is about to be added to history.

**Comment by zimulala** [comment]
PTAL @coocood @tiancaiamao @shenli @GregoryIan

**Comment by coocood** [comment]
LGTM

**Comment by GregoryIan** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**